sled-agent/src/sim/storage.rs

```diff
@@ -1172,10 +1175,15 @@ impl Zpool {
                 end_port,
             )),
         );
+
         let DatasetContents::Crucible(crucible) = self
             .datasets
             .get(&id)
-            .expect("Failed to get the dataset we just inserted");
+            .expect("Failed to get the dataset we just inserted")
```

Comment:
Untested, but I think we could get rid of this expect by replacing the `insert()` / `get()` with `entry()` / `insert_entry()`:

```rust
let DatasetContents::Crucible(crucible) = self
    .datasets
    .entry(id)
    .insert_entry(DatasetContents::Crucible(/* ... blah blah ...*/))
    .into_mut()
else {
    unreachable!("...");
};
```

 (but +1 on @hawkw's suggestion for `unreachable!()` over `panic!()`)